src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexWriter.cs

```diff
@@ -56,7 +53,7 @@ internal static RegexCode Write(RegexTree t)
         private RegexWriter()
         {
             _intStack = new int[32];
-            _emitted = new int[32];
+            _emitted = new List<int>(32);
```

Comment:
I think we should use a larger size here to avoid the cost of resizing for long patterns. E.g. 50 would better a fit and would only increase memory memory by 72 byte

Reply:
Whatever you do should be supported by measurements...

Reply:
> to avoid the cost of resizing for long patterns

Do we have any data on the distribution of pattern lengths?

Reply:
> Average is about 58 op codes.

What data set is this based on?

Reply:
If there's a correlation between pattern length and opcode count you could possibly use a multiplier of some sort.

Reply:
Because of character sets and "multis" (string text) in most patterns, there is no strong correlation...